src/main/java/bq_standard/tasks/ProgressUtil.java

```diff
@@ -0,0 +1,58 @@
+package bq_standard.tasks;
+
+
+import betterquesting.api2.utils.DirtyPlayerMarker;
+
+import javax.annotation.Nullable;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+
+public class ProgressUtil {
```

Comment:
Shouldn't these be extracted to something like an AbstractTask base class instead of moving everything off to a utility class?

Reply:
Ok
I just didn't expect all Tank's structure to be so uniform at first

Reply:
Modified